Mathlib/Data/Nat/Choose/Lucas.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2023 Gareth Ma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gareth Ma
+-/
+import Mathlib.Data.ZMod.Basic
+import Mathlib.RingTheory.Polynomial.Basic
+
+/-!
+# Lucas's theorem
+
+This file contains a proof of [Lucas's theorem](https://www.wikiwand.com/en/Lucas%27s_theorem) about
+binomial coefficients, which says that for primes `p`, `n` choose `k` is congruent to product of
+`n_i` choose `k_i` modulo `p`, where `n_i` and `k_i` are the base-`p` digits of `n` and `k`,
+respectively.
+
+## Main statements
+
+* `lucas_theorem''`: the binomial coefficient `n choose k` is congruent to the
```

Comment:
```suggestion
* `choose_modEq_prod_range_choose`: the binomial coefficient `n choose k` is congruent to the
```

Reply:
@Parcly-Taxel, please don't merge in your own suggestions on someone else's PR when they are potentially contentious